### PR TITLE
Add compress command

### PR DESCRIPTION
**Pull request by xla**
This command is intended to be used on large journals to reduce the time needed to restore from it. It reads a journal and keeps it in-memory for every DEL operation occuring it removes the entry and overwrites when a SET occurs and the path already is present, after reading the entire journal and only storing the necessary paths and their values it dumps the state to the file given on command invocation.

***
@soundcloud/iss

### CODE REVIEW

**Comment by stuartnelson3** [comment]
playing the part of tobi... got any tests?

File: compress.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		case journal.OpSet:
+			tree[entry.Path] = entry
+		case journal.OpDel:
+			delete(tree, entry.Path)
+		default:
+			return fmt.Errorf("unknown operation %s", entry.Op)
+		}
+	}
+
+	f, err = os.Create(cmd.Args[0])
```
Comment (grobie) [inline compress.go]: I'd vote to not reuse the same variable name for different file handles.

File: compress.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func runCompress(cmd *Command) error {
+	if len(cmd.Args) != 1 {
+		cmd.Usage()
+	}
+
+	tree := map[string]*journal.Entry{}
+
+	f, err := os.OpenFile(cmd.Globals.File, os.O_RDONLY, 0666)
```
Comment (grobie) [inline compress.go]: Why do you need the lower level `OpenFile` here instead of `Open`? What do these permissions do for reading?

**Comment by grobie** [comment]
Great!

Yeah, a high level test would be great. One idea: 1) take "dirty" journal -> load into doozer -> dump 2) take "dirty" jounral -> compress -> load into doozerd -> dump 3) compare 1) and 2).

I'd also be interested to see numbers about the impact of compression, probably added to the readme.

**Pull request merged**